src/loaders/ObjectLoader.js

```diff
@@ -323,44 +316,16 @@ class ObjectLoader extends Loader {
 			const loader = new MaterialLoader();
 			loader.setTextures( textures );
 
-			for ( let i = 0, l = json.length; i < l; i ++ ) {
-
-				const data = json[ i ];
-
-				if ( data.type === 'MultiMaterial' ) {
-
-					// Deprecated
-
-					const array = [];
+			for ( const data of json ) {
 
-					for ( let j = 0; j < data.materials.length; j ++ ) {
+				if ( cache[ data.uuid ] === undefined ) {
 
-						const material = data.materials[ j ];
-
-						if ( cache[ material.uuid ] === undefined ) {
-
-							cache[ material.uuid ] = loader.parse( material );
-
-						}
-
-						array.push( cache[ material.uuid ] );
-
-					}
-
-					materials[ data.uuid ] = array;
-
-				} else {
-
-					if ( cache[ data.uuid ] === undefined ) {
```

Comment:
Why did you remove this `if` statement?

Reply:
It is said there that it is deprecated. Or is it talking about the entire multi-materials usage?

Reply:
Only the `MultiMaterial` branch is deprecated.

Reply:
I think I've removed only it? The other branch remains how it was.

Reply:
Ah, the `if ( cache[ data.uuid ] === undefined ) {` is further above 👍 .